### PR TITLE
[Behat][Promotion] Percentage discount on items in range

### PR DESCRIPTION
**Pull request by Zales0123**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| Related tickets |
| License         | MIT
| Doc PR          |

### CODE REVIEW

File: src/Sylius/Component/Core/Promotion/Action/ItemDiscountAction.php [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        if (!$subject instanceof OrderInterface) {
+            throw new UnexpectedTypeException($subject, OrderInterface::class);
+        }
+
+        foreach ($subject->getItems() as $item) {
+            $this->removeUnitsAdjustment($item, $promotion);
+        }
+    }
+
+    public abstract function getConfigurationFormType();
```
Comment (NeverResponse) [inline src/Sylius/Component/Core/Promotion/Action/ItemDiscountAction.php]: Doc blok missing

**Comment by pjedrzejewski** [comment]
@Zales0123 Requires rebasing.

File: src/Sylius/Behat/Context/Setup/PromotionContext.php [hunk 2]
```diff
@@ -262,9 +289,28 @@ private function configureActionTaxonFilter(ActionInterface $action, array $taxo
...
+    {
+        $this->persistPromotionWithAction($promotion, $this->actionFactory->createItemFixedDiscount($discount), $configuration);
+    }
+
+    /**
+     * @param PromotionInterface $promotion
+     * @param int $discount
+     * @param array $configuration
+     */
+    private function createPercentagePromotionWithPriceRangeFilter(PromotionInterface $promotion, $discount, array $configuration)
```
Comment (lchrusciel) [inline src/Sylius/Behat/Context/Setup/PromotionContext.php]: Can we split it?

Comment (lchrusciel) [inline src/Sylius/Behat/Context/Setup/PromotionContext.php]: Anyway, Is there any need for such method? We reduce this call `$this->actionFactory->createItemPercentageDiscount($discount)` and put whole new method for this. I think it is not worth it.

**Comment by pjedrzejewski** [comment]
Thank you Mateusz!

**Pull request merged**